### PR TITLE
Code style Update

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### What does this implement/fix? Explain your changes.
This PR is a trial for enhancing the repositories' code style. Below is provided some overall changes:
+ Unifying code style in documentation, code, and tests
+ `__class_stat_init__` function in `pycm_handler.py` modified
+ `__overall_stat_init__` function in `pycm_handler.py`
+ `ConfusionMatrix.__init__` method
+ extra checks in `pycm_interpret.py`
+ `save_obj` modified 

_Note: This might look like a huge PR, but reviewing it will take no time since it's all style-related._

### CODE REVIEW

File: pycm/pycm_handler.py [hunk 1]
```diff
@@ -88,73 +31,24 @@ def __overall_stat_init__(cm):
...
-    cm.Alpha = cm.overall_stat["Krippendorff Alpha"]
-    cm.SOA7 = cm.overall_stat["SOA7(Lambda A)"]
-    cm.SOA8 = cm.overall_stat["SOA8(Lambda B)"]
-    cm.SOA9 = cm.overall_stat["SOA9(Krippendorff Alpha)"]
-    cm.SOA10 = cm.overall_stat["SOA10(Pearson C)"]
+    for stat, field_name in OVERALL_PARAMS2FIELD_NAME.items():
+        setattr(cm, field_name, cm.overall_stat[stat])
+
+
+def __imbalacement_handler__(cm, is_imbalanced):
```
Comment (sepandhaghighi) [inline pycm/pycm_handler.py:38]: imbalancement

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 ### Added
+- `__imbalacement_handler__` function
```
Comment (sepandhaghighi) [inline CHANGELOG.md:9]: imbalancement

File: pycm/pycm_obj.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 from .pycm_error import pycmVectorError, pycmMatrixError, pycmCIError, pycmAverageError, pycmPlotError
 from .pycm_handler import __class_stat_init__, __overall_stat_init__
 from .pycm_handler import __obj_assign_handler__, __obj_file_handler__, __obj_matrix_handler__, __obj_vector_handler__, __obj_array_handler__
+from .pycm_handler import __imbalacement_handler__
```
Comment (sepandhaghighi) [inline pycm/pycm_obj.py:7]: imbalancement

File: pycm/pycm_handler.py [hunk 4]
```diff
@@ -88,73 +31,24 @@ def __overall_stat_init__(cm):
...
-    cm.SOA6 = cm.overall_stat["SOA6(Matthews)"]
-    cm.CSI = cm.overall_stat["CSI"]
-    cm.ARI = cm.overall_stat["ARI"]
-    cm.B = cm.overall_stat["Bangdiwala B"]
-    cm.Alpha = cm.overall_stat["Krippendorff Alpha"]
-    cm.SOA7 = cm.overall_stat["SOA7(Lambda A)"]
-    cm.SOA8 = cm.overall_stat["SOA8(Lambda B)"]
-    cm.SOA9 = cm.overall_stat["SOA9(Krippendorff Alpha)"]
-    cm.SOA10 = cm.overall_stat["SOA10(Pearson C)"]
+    for stat, field_name in OVERALL_PARAMS2FIELD_NAME.items():
```
Comment (sepandhaghighi) [inline pycm/pycm_handler.py:34]: I have a suggestion for this part:

1. We already have a `CLASS_PARAMS` in `pycm_params.py`, convert it to a dictionary that map `name` to `value` (they are the same at this moment)
2. Rename  `OVERALL_PARAMS2FIELD_NAME` to `OVERALL_PARAMS`
3. We need to update wherever these variables are used
4. We need to update `CONTRIBUTING.md`

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

**Review by alirezazolanvari — APPROVED** [review]
@sadrasabouri It's simply impressive. Thanks for your effort.

**Review by sepandhaghighi — APPROVED** [review]
LGTM 🚀

**Pull request merged**